pkg/controllers/kube-apiserver.go

```diff
@@ -262,14 +260,15 @@ func (s *KubeAPIServer) Run(ctx context.Context, ready chan<- struct{}, stopped
 	// run readiness check
 	go func() {
 		err := wait.PollImmediateWithContext(ctx, time.Second, kubeAPIStartupTimeout*time.Second, func(ctx context.Context) (bool, error) {
-			restConfig, err := clientcmd.BuildConfigFromFlags("", s.kubeconfig)
+			restConfig, err := clientcmd.BuildConfigFromFlags(s.masterURL, "")
 			if err != nil {
 				return false, err
 			}
 			if err := rest.SetKubernetesDefaults(restConfig); err != nil {
 				return false, err
 			}
 			restConfig.NegotiatedSerializer = serializer.NewCodecFactory(runtime.NewScheme())
+			restConfig.CAFile = s.servingCertPath
```

Comment:
Should we use the CA from https://github.com/openshift/microshift/blob/main/pkg/controllers/kube-apiserver.go#L96?

Reply:
Since this is just a startup test and we are specifically interested in checking this very server, I believe a simple trust such as we've got here is adequate